### PR TITLE
Tiles

### PR DESCRIPTION
**Pull request by jordankoschei-okta**

### CODE REVIEW

File: packages/odyssey-react-mui/src/Button.tsx [hunk 1]
```diff
@@ -104,14 +110,22 @@ export type ButtonProps = {
 ) &
   SeleniumProps;
 
+export type ButtonContextValue = {
+  isFullWidth?: boolean;
+};
+
+export const ButtonContext = createContext<ButtonContextValue>({
+  isFullWidth: undefined,
+});
```
Comment (KevinGhadyani-Okta) [inline packages/odyssey-react-mui/src/Button.tsx:119]: We can default this to `false`:

```suggestion
export type ButtonContextValue = {
  isFullWidth?: boolean;
};

export const ButtonContext = createContext<ButtonContextValue>({
  isFullWidth: false,
});
```

File: packages/odyssey-react-mui/src/Button.tsx [hunk 2]
```diff
@@ -151,7 +167,7 @@ const Button = ({
       endIcon,
       id,
       isDisabled,
-      isFullWidth,
+      isFullWidthProp,
```
Comment (KevinGhadyani-Okta) [inline packages/odyssey-react-mui/src/Button.tsx:170]: This is missing `isFullWidth`.

File: packages/odyssey-react-mui/src/MenuButton.tsx [hunk 3]
```diff
@@ -49,10 +62,12 @@ export type MenuButtonProps = {
   /**
    * The <MenuItem> components within the Menu.
    */
-  children: Array<
-    | ReactElement<typeof MenuItem | typeof Divider | typeof ListSubheader>
-    | NullElement
-  >;
+  children:
+    | ReactFragment
```
Comment (jordankoschei-okta) [inline packages/odyssey-react-mui/src/MenuButton.tsx:66]: Add generic if possible

File: packages/odyssey-react-mui/src/MenuButton.tsx [hunk 4]
```diff
@@ -65,6 +80,18 @@ export type MenuButtonProps = {
    * If the MenuButton is an overflow menu or standard menu.
    */
   isOverflow?: boolean;
+  /**
+   * The horizontal alignment of the menu.
+   */
+  menuAlignment?: "left" | "right";
```
Comment (KevinGhadyani-Okta) [inline packages/odyssey-react-mui/src/MenuButton.tsx:86]: Don't these need to be in an array for Storybook?

File: packages/odyssey-react-mui/src/MenuButton.tsx [hunk 5]
```diff
@@ -110,11 +140,13 @@ const MenuButton = ({
 
   const isOpen = Boolean(anchorEl);
 
-  const closeMenu = useCallback<MenuContextType["closeMenu"]>(() => {
+  const closeMenu = useCallback((event, reason) => {
+    onClose?.(event, reason);
     setAnchorEl(null);
   }, []);
```
Comment (KevinGhadyani-Okta) [inline packages/odyssey-react-mui/src/MenuButton.tsx:146]: Callback missing `onClose`.

File: packages/odyssey-react-mui/src/MenuButton.tsx [hunk 6]
```diff
@@ -110,11 +140,13 @@ const MenuButton = ({
...
 
-  const openMenu = useCallback<MenuContextType["openMenu"]>((event) => {
+  const openMenu = useCallback((event) => {
+    onClick?.(event);
     setAnchorEl(event.currentTarget);
   }, []);
```
Comment (KevinGhadyani-Okta) [inline packages/odyssey-react-mui/src/MenuButton.tsx:151]: Missing `onClose` in dependency arg.

File: packages/odyssey-react-mui/src/MenuButton.tsx [hunk 7]
```diff
@@ -162,6 +194,8 @@ const MenuButton = ({
 
       <Menu
         anchorEl={anchorEl}
+        anchorOrigin={{ horizontal: menuAlignment, vertical: "bottom" }}
+        transformOrigin={{ horizontal: menuAlignment, vertical: "top" }}
```
Comment (KevinGhadyani-Okta) [inline packages/odyssey-react-mui/src/MenuButton.tsx:198]: Can we memoize these values?

File: packages/odyssey-react-mui/src/Tile.tsx [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+      onClick: () => void;
+    }
+  | {
+      button: ReactElement<typeof Button> | NullElement;
+      onClick?: () => void;
+    }
+  | {
+      button: undefined | null;
+      onClick: undefined | null;
+    }
```
Comment (KevinGhadyani-Okta) [inline packages/odyssey-react-mui/src/Tile.tsx:46]: I wonder if we can change these to `never`?

File: packages/odyssey-react-mui/src/Tile.tsx [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+      <>
+        {image && (
+          <Box
+            sx={{
+              display: "flex",
+              alignItems: "flex-start",
+              maxHeight: "64px",
+              marginBlockEnd: 5,
+              paddingRight: menuItems ? "20px" : 0,
+            }}
```
Comment (KevinGhadyani-Okta) [inline packages/odyssey-react-mui/src/Tile.tsx:74]: Can we `useMemo` these?

Comment (KevinGhadyani-Okta) [reply packages/odyssey-react-mui/src/Tile.tsx:74]: Ideally, these styles would be part of the theme.

File: packages/odyssey-react-mui/src/Tile.tsx [hunk 10]
```diff
@@ -0,0 +1,133 @@
...
+
+        {overline && <Support component="div">{overline}</Support>}
+        {title && <Heading5 component="div">{title}</Heading5>}
+        {description && (
+          <Paragraph color="textSecondary">{description}</Paragraph>
+        )}
+
+        {button && (
+          <MuiCardActions>
+            <ButtonContext.Provider value={{ isFullWidth: true }}>
```
Comment (KevinGhadyani-Okta) [inline packages/odyssey-react-mui/src/Tile.tsx:88]: Can we set this context at the top of the component above the function, so it's a static memory refrence?

`const buttonContextValue`.

File: packages/odyssey-react-mui/src/Tile.tsx [hunk 11]
```diff
@@ -0,0 +1,133 @@
...
+    );
+  }, [button, description, image, menuItems, overline, title]);
+
+  return (
+    <MuiCard className={onClick ? "isClickable" : ""}>
+      {onClick && (
+        <MuiCardActionArea onClick={onClick}>{cardContent}</MuiCardActionArea>
+      )}
+
+      {!onClick && cardContent}
```
Comment (KevinGhadyani-Okta) [inline packages/odyssey-react-mui/src/Tile.tsx:103]: This can be a ternary.

File: packages/odyssey-react-mui/src/Tile.tsx [hunk 12]
```diff
@@ -0,0 +1,133 @@
...
+        <Box
+          sx={{
+            position: "absolute",
+            right: "8px",
+            top: "8px",
+          }}
+        >
+          <MenuButton
+            endIcon={<MoreIcon />}
+            ariaLabel="Tile menu"
```
Comment (KevinGhadyani-Okta) [inline packages/odyssey-react-mui/src/Tile.tsx:115]: Can we modify this label?

File: packages/odyssey-react-mui/src/Tile.tsx [hunk 13]
```diff
@@ -0,0 +1,133 @@
...
+
+      {!onClick && cardContent}
+
+      {menuItems && (
+        <Box
+          sx={{
+            position: "absolute",
+            right: "8px",
+            top: "8px",
+          }}
```
Comment (KevinGhadyani-Okta) [inline packages/odyssey-react-mui/src/Tile.tsx:111]: Can we memoize these?

File: packages/odyssey-react-mui/src/properties/ts/.ts.DS_Store [hunk 14]
```diff
@@ -0,0 +1 @@
...000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000"};
```
Comment (KevinGhadyani-Okta) [inline packages/odyssey-react-mui/src/properties/ts/.ts.DS_Store:1]: Remove this file and add to `.gitignore`.

`git rm --cached packages/odyssey-react-mui/src/properties/ts/.ts.DS_Store`

File: packages/odyssey-react-mui/src/theme/components.tsx [hunk 15]
```diff
@@ -582,6 +582,69 @@ export const components = ({
...
+          },
+        }),
+      },
+    },
+    MuiCardActionArea: {
+      styleOverrides: {
+        root: () => ({
+          margin: `-${odysseyTokens.Spacing5}`,
+          padding: odysseyTokens.Spacing5,
+          width: `calc(100% + (${odysseyTokens.Spacing5} * 2))`,
```
Comment (KevinGhadyani-Okta) [inline packages/odyssey-react-mui/src/theme/components.tsx:628]: Let's add a ticket to fix this.

File: packages/odyssey-react-mui/src/theme/components.tsx [hunk 16]
```diff
@@ -582,6 +582,69 @@ export const components = ({
...
+      styleOverrides: {
+        root: () => ({
+          borderRadius: odysseyTokens.BorderRadiusOuter,
+          boxShadow: odysseyTokens.DepthDialog,
+          padding: odysseyTokens.Spacing5,
+          position: "relative",
+
+          "& img": {
+            height: "64px",
+          },
```
Comment (KevinGhadyani-Okta) [inline packages/odyssey-react-mui/src/theme/components.tsx:595]: Can this a direct child of the root?

`& > img`?

File: packages/odyssey-storybook/src/components/odyssey-labs/Tile/Tile.stories.tsx [hunk 17]
```diff
@@ -0,0 +1,146 @@
...
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { Meta, StoryObj } from "@storybook/react";
+import { MuiThemeDecorator } from "../../../../.storybook/components";
+import { Box, Button, Callout, Tile, MenuItem } from "@okta/odyssey-react-mui";
+
+const storybookMeta: Meta = {
```
Comment (KevinGhadyani-Okta) [inline packages/odyssey-storybook/src/components/odyssey-labs/Tile/Tile.stories.tsx:17]:
```suggestion
const storybookMeta: Meta<TileProps> = {
```

File: packages/odyssey-storybook/src/components/odyssey-mui/MenuButton/MenuButton.stories.tsx [hunk 18]
```diff
@@ -133,6 +133,16 @@ const storybookMeta: Meta<MenuButtonProps> = {
...
       },
     },
+    menuAlignment: {
+      options: ["left", "right"],
+      control: { type: "radio" },
+      description: "The horizontal alignment of the menu.",
+      table: {
+        type: {
+          summary: "left | right",
+        },
```
Comment (KevinGhadyani-Okta) [inline packages/odyssey-storybook/src/components/odyssey-mui/MenuButton/MenuButton.stories.tsx:143]: "left" and "right" should be in an array/union type.

**Review by KevinGhadyani-Okta — CHANGES_REQUESTED** [review]
A bunch of small comments.

**Comment by jordankoschei-okta** [comment]
Closing because this is stale and needs some serious rework

**Pull request closed without merging**